src/Components/test/Microsoft.AspNetCore.Components.E2ETest/ServerExecutionTests/ServerComponentRenderingTest.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using BasicTestApp;
+using Microsoft.AspNetCore.Components.Browser.Rendering;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Components.E2ETest.Tests;
+using OpenQA.Selenium;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.ServerExecutionTests
+{
+    // By inheriting from ComponentRenderingTest, this test class also copies
+    // all the test cases shared with client-side rendering
+
+    public class ServerComponentRenderingTest : ComponentRenderingTest
+    {
+        public ServerComponentRenderingTest(BrowserFixture browserFixture, ToggleExecutionModeServerFixture<Program> serverFixture, ITestOutputHelper output)
+            : base(browserFixture, serverFixture.WithServerExecution(), output)
+        {
+        }
+
+        [Fact]
+        public void ThrowsIfRenderIsRequestedOutsideSyncContext()
```

Comment:
There's no WebAssembly equivalent to this exception. However code running under WebAssembly can still use `Invoke` and `InvokeAsync` and will get the same behavior as on the server, so the same components still work in both places (assuming you don't *rely* on causing sync context exceptions).